src/transformers/modeling_utils.py

```diff
@@ -473,14 +473,12 @@ def load_state_dict(checkpoint_file: Union[str, os.PathLike]):
             )
         return safe_load_file(checkpoint_file)
     try:
-        if (
-            (is_deepspeed_zero3_enabled() or is_fsdp_enabled())
-            and torch.distributed.is_initialized()
-            and torch.distributed.get_rank() > 0
-        ):
+        if (is_deepspeed_zero3_enabled()) and torch.distributed.is_initialized() and torch.distributed.get_rank() > 0:
             map_location = "meta"
         else:
             map_location = "cpu"
+
+        map_location = "cpu" if is_fsdp_enabled_and_local_dist_rank_0 else "meta"
```

Comment:
```suggestion
        map_location = "cpu" if is_fsdp_enabled_and_local_dist_rank_0() else "meta"
```

Also, wouldn't it be more readable to integrate this an `elif` to the conditional above?

Reply:
> Also, wouldn't it be more readable to integrate this an elif to the conditional above?

done